src/Refactoring-UI/RBMakeClassAbstractDriver.class.st

```diff
@@ -0,0 +1,46 @@
+Class {
+	#name : 'RBMakeClassAbstractDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'class'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'execution' }
+RBMakeClassAbstractDriver >> changes [
+	"Remember should not call generateChanges"
+
+	refactoring privateTransform.
+	^ refactoring changes
+]
+
+{ #category : 'resources' }
+RBMakeClassAbstractDriver >> configureRefactoring [
+	"Configure the transformation"
+
+	refactoring := RBMakeClassAbstractTransformation new
+		model: model;
+		classNamed: class name;
+		yourself.
+]
+
+{ #category : 'execution' }
+RBMakeClassAbstractDriver >> runRefactoring [
+	"Run the transformation"
+
+	self configureRefactoring.
+	refactoring failedApplicabilityPreconditions
+		ifNotEmpty: [ ^ self inform: 'Preconditions not met' ].
```

Comment:
Can we here tell the user what exactly went wrong? We know there is only one precontiion, and `refactoring failedApplicabilityPreconditions` returns a list with only that precondition. I'm suggesting that we do something like

```suggestion
	preconditions := refactoring failedApplicabilityPreconditions
	preconditions
		ifNotEmpty: [ ^ self inform: preconditions first errorMessage ].
```
Maybe the code needs to be modified a bit, but you get the gist of it.